instrumentation/opencensus-shim/testing/src/test/java/io/opentelemetry/opencensusshim/JavaagentInstrumentationTest.java

```diff
@@ -0,0 +1,339 @@
+        try (io.opencensus.common.Scope innerScope = ocTracer.withSpan(innerSpan)) {
+        } finally {
+          innerSpan.end();
+        }
+      } finally {
+        midSpan.end();
+      }
+    } finally {
+      outerSpan.end();
+    }
+
+    Tracing.getExportComponent().shutdown();
+
+    SpanContext outerContext = SpanConverterProxy.mapSpanContext(outerSpan.getContext());
+    SpanContext midContext = SpanConverterProxy.mapSpanContext(midSpan.getContext());
+
+    // expecting 1 trace with 3 spans
+    testing.waitAndAssertTraces(
+        ta ->
+            // ensure each span's attributes haven't seeped into parents or children
+            ta.hasSpansSatisfyingExactly(
+                // inner span
+                sa ->
+                    sa.hasName("inner-span")
+                        .hasParentSpanId(midContext.getSpanId())
+                        .hasAttribute(AttributeKey.booleanKey("inner"), true)
+                        .hasAttributesSatisfying(
+                            OpenTelemetryAssertions.satisfies(
+                                AttributeKey.booleanKey("middle"), AbstractBooleanAssert::isNull),
+                            OpenTelemetryAssertions.satisfies(
+                                AttributeKey.booleanKey("outer"), AbstractBooleanAssert::isNull)),
+                // middle span
+                sa ->
+                    sa.hasName("mid-span")
+                        .hasParentSpanId(outerContext.getSpanId())
+                        .hasAttribute(AttributeKey.booleanKey("middle"), true)
+                        .hasAttributesSatisfying(
+                            OpenTelemetryAssertions.satisfies(
+                                AttributeKey.booleanKey("inner"), AbstractBooleanAssert::isNull),
+                            OpenTelemetryAssertions.satisfies(
+                                AttributeKey.booleanKey("outer"), AbstractBooleanAssert::isNull)),
+                // outer span
+                sa ->
+                    sa.hasName("outer-span")
+                        .hasNoParent()
+                        .hasAttribute(AttributeKey.booleanKey("outer"), true)
+                        .hasAttributesSatisfying(
+                            OpenTelemetryAssertions.satisfies(
+                                AttributeKey.booleanKey("inner"), AbstractBooleanAssert::isNull),
+                            OpenTelemetryAssertions.satisfies(
+                                AttributeKey.booleanKey("middle"),
+                                AbstractBooleanAssert::isNull))));
+  }
+
+  static io.opencensus.trace.Span findSampledOCSpan(
+      io.opencensus.trace.Tracer ocTracer, String name) {
+    // loop until we get a span which will be exported (isSampled);
+    // opencensus-shim at its root uses the default sampler which is probability-based,
+    // and we need the span to be emitted in order to adequately test;
```

Comment:
the default sampling percentage is not 100%? if not, can we configure it in the test to be 100%?

Reply:
Unfortunately, this is a hard-coded sampling rate, rooted [here](https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/trace/config/TraceParams.java#L39). There might be a way around it...let me take a closer look and report back.

Reply:
OK, looks like we might be able to set the `TraceParams` for the test. I can't fully test this until https://github.com/open-telemetry/opentelemetry-java/pull/4900 is merged. Let me address your comments there to keep that moving, but looks like that one is good otherwise?

Reply:
I've wired up a `setup()` method which I think will work. Once the dependent PR is merged, I can verify this method is no longer needed and remove it.

As an aside though FFR: we might need to look into how opencensus-shim is handling the propagation of otel sampling down to opencensus sampling bc I suspect there may be a disconnect, simple point in fact being I needed to make this method in the first place. Where/how would you like to track that? If at all?

Reply:
> we might need to look into how opencensus-shim is handling the propagation of otel sampling down to opencensus sampling bc I suspect there may be a disconnect, simple point in fact being I needed to make this method in the first place

can you check the OpenCensus Compatibility spec to see if this is mentioned/covered? https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/opencensus.md

cc @dashpole